pkg/state/service_test.go

```diff
@@ -104,6 +104,19 @@ func TestServicesCollection_Add(t *testing.T) {
 	}
 }
 
+func TestServiceInsertIgnoreDuplicate(t *testing.T) {
+	assert := assert.New(t)
+	collection := servicesCollection()
+
+	var s Service
+	s.ID = kong.String("my-id")
+	s.Name = kong.String("first")
+	err := collection.Add(s)
+	assert.Nil(err)
+	err = collection.AddIgnoringDuplicates(s)
+	assert.Nil(err)
```

Comment:
Food for thought. ( but as I can see that's currently done in all the tests added in this PR so ideally if this is changed we'd do it everywhere for consistency).

```suggestion
	collection := servicesCollection()

	var s Service
	s.ID = kong.String("my-id")
	s.Name = kong.String("first")
	require.NoError(t, collection.Add(s))
	require.NoError(t, collection.AddIgnoringDuplicates(s))
```

Reply:
Updated in all tests